src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -0,0 +1,236 @@
+        SearchList.insert(std::pair<IInspectable, hstring>(Windows::Foundation::PropertyValue::CreateString(L"Screen redrawing"), L"Rendering_Nav"));
+        SearchList.insert(std::pair<IInspectable, hstring>(Windows::Foundation::PropertyValue::CreateString(L"Show terminal title in title bar"), L"GlobalAppearance_Nav"));
+        SearchList.insert(std::pair<IInspectable, hstring>(Windows::Foundation::PropertyValue::CreateString(L"Show the title bar"), L"GlobalAppearance_Nav"));
+        SearchList.insert(std::pair<IInspectable, hstring>(Windows::Foundation::PropertyValue::CreateString(L"Software rendering"), L"Rendering_Nav"));
+        SearchList.insert(std::pair<IInspectable, hstring>(Windows::Foundation::PropertyValue::CreateString(L"Tab width mode"), L"GlobalAppearance_Nav"));
+        SearchList.insert(std::pair<IInspectable, hstring>(Windows::Foundation::PropertyValue::CreateString(L"Theme"), L"GlobalAppearance_Nav"));
+        SearchList.insert(std::pair<IInspectable, hstring>(Windows::Foundation::PropertyValue::CreateString(L"Window resize behavior"), L"Rendering_Nav"));
+        SearchList.insert(std::pair<IInspectable, hstring>(Windows::Foundation::PropertyValue::CreateString(L"Word delimeters"), L"Interaction_Nav"));
+    }
+
+    void MainPage::ClickHandler(IInspectable const&, RoutedEventArgs const&)
+    {
+    }
+
+    void MainPage::SettingsNav_SelectionChanged(const MUX::Controls::NavigationView, const MUX::Controls::NavigationViewSelectionChangedEventArgs)
+    {
+    }
+
+    void MainPage::SettingsNav_Loaded(IInspectable const&, RoutedEventArgs const&)
+    {
+        //// set the initial selectedItem
+        for (uint32_t i = 0; i < SettingsNav().MenuItems().Size(); i++)
+        {
+            const auto item = SettingsNav().MenuItems().GetAt(i).as<Controls::ContentControl>();
+            const hstring homeNav = L"Home_Nav";
+            const hstring itemTag = unbox_value<hstring>(item.Tag());
+
+            if (itemTag == homeNav)
+            {
+                SettingsNav().SelectedItem(item);
+                break;
+            }
+        }
+
+        contentFrame().Navigate(xaml_typename<Editor::Home>());
+    }
+
+    void MainPage::SettingsNav_ItemInvoked(MUX::Controls::NavigationView const&, MUX::Controls::NavigationViewItemInvokedEventArgs const& args)
+    {
+        auto clickedItemContainer = args.InvokedItemContainer();
+
+        if (clickedItemContainer != NULL)
+        {
+            Navigate(contentFrame(), unbox_value<hstring>(clickedItemContainer.Tag()));
+        }
+    }
+
+    void MainPage::SettingsNav_BackRequested(MUX::Controls::NavigationView const&, MUX::Controls::NavigationViewBackRequestedEventArgs const& /*args*/)
+    {
+        On_BackRequested();
+    }
+
+    bool MainPage::On_BackRequested()
+    {
+        if (!contentFrame().CanGoBack())

```

Comment:
Not sure if this is the only place, but we generally don't do single line if statements without braces...